### DISCUSSION TITLE
[Feature Request] Allow specified actors to bypass required status checks

### DISCUSSION THREAD

**Discussion by lclausen-r7** [community/community#18145] [Pull Requests]
Similar to the branch setting `Allow specified actors to bypass required pull requests`, it would be very helpful to have similar granular control for status checks where specified accounts can merge without status checks passing. This would be helpful for automating tasks within repositories

**Comment by mightyiam** [comment]
A workaround is to push to a temporary branch, wait for the checks to pass and then push to the protected branch.
Of course, this feature would be much nicer.

**Reply by PascalBourdier** [reply]
Ok but you have to create PR to trigger checks ?

**Reply by mightyiam** [reply]
@PascalBourdier if the workflows that are marked as your checks are triggered by merely pushing then no PR is necessary.

**Comment by esran** [comment]
This would be super useful for things like automated version bump of repositories without having to force it through on an admin account. Nobody wants to give automated tools admin!

**Reply by natanans** [reply]
I cant agree more!

**Comment by sricks** [comment]
This could be useful, but we also need granularity over which status checks can be bypassed. Not having this granularity encourages bad security practices.

Imagine if you have one status check "virus-free" and one status check "function declarations are alphabetized". We want to ensure 100% of branches are virus-free, but engineers may need to deploy a PR without alphabetized function declarations (or whatever other silly requirement you may think of).

There's no way to disable one without the other.

**Reply by mightyiam** [reply]
This feature without selective status checks would be better than not having this feature. Not having this feature already encourages workarounds that are less secure.

**Comment by johnjelinek** [comment]
This feature would allow terraform to use a service account user to push files directly to github repos without going through the whole PR flow. This is a much needed feature.

**Comment by Colin-b** [comment]
Much needed feature, having to wait for a runner to perform checks (actual one or dummy one with the same name/id) is not a proper option, so is temporary deactivating those required status checks.

**Comment by justinmchase** [comment]
Looks like a duplicate of this discussion: https://github.com/orgs/community/discussions/43460

I too need this for a version incrementing task. It definitely seems legitimate to me for certain automated tasks to be able to commit directly to a protected branch without status checks.

**Comment by rfkd** [comment]
Need that, too. Any update?

**Reply by na-jakobs** [reply]
+1

**Comment by na-jakobs** [comment]
Much needed feature 🙏

**Comment by justinmchase** [comment]
It looks like the new "ruleset" features will support this. I haven't tested it out yet though.

**Reply by justinmchase** [reply]
I tested it out finally, by the way, and it solves it all perfectly. Great implementation, thank you all!

**Comment by raihle** [comment]
The Ruleset / Rules feature mentioned above (currently in public beta) does fix this! 🎉

**Reply by johnjelinek** [reply]
link?

**Reply by neotrow** [reply]
Can also confirm that this works for our use case 🎉  

@johnjelinek https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets

**Comment by ThePeterDenham** [comment]
Please skip checks based on the commit message or a label. Then I can skip the automated tests on small document changes and merge even though they are required and have not been run.

**Reply by justinmchase** [reply]
I don't think there is a way to do that safely.

**Comment by rfkd** [comment]
When will rulesets be available in GHES?

**Reply by coopernetes** [reply]
According to the public roadmap, it's scheduled for GHES 3.11: 
https://github.com/orgs/github/projects/4247/views/5?filterQuery=label%3Aserver+-status%3AWelcome%21++repository&pane=issue&itemId=21176211

**Comment by vegeta321311** [comment]
0x2e7df6483c7e1d2ee6cb11ac70d37fb060cf6e23
